test/c3/appframework/appframework_test.go

```diff
@@ -211,4 +212,102 @@ var _ = Describe("c3appfw test", func() {
 			testenv.VerifyAppsCopied(deployment, testenvInstance, testenvInstance.GetName(), podNames, appListV1, false, true)
 		})
 	})
+
+	Context("Clustered deployment (C3 - clustered indexer, search head cluster)", func() {
+		It("c3, appframework: can deploy a C3 SVA and have ES app installed on SHC", func() {
+
+			// ES is a huge file, we configure it here rather than in BeforeSuite/BeforeEach to save time for other tests
+			// Upload ES app to S3
+			esApp := "SplunkEnterpriseSecuritySuite"
+			appListV1 = append(testenv.BasicApps, esApp)
+			appFileList := testenv.GetAppFileList(appListV1, 1)
+
+			// Download V1 Apps from S3
+			err := testenv.DownloadFilesFromS3(testDataS3Bucket, s3AppDirV1, downloadDirV1, appFileList)
+			Expect(err).To(Succeed(), "Unable to download V1 app files")
+
+			// Upload V1 apps to S3
+			uploadedFiles, err := testenv.UploadFilesToS3(testS3Bucket, s3TestDir, appFileList, downloadDirV1)
+			Expect(err).To(Succeed(), "Unable to upload ES app to S3 test directory")
+			uploadedApps = append(uploadedApps, uploadedFiles...)
+
+			// Create App framework Spec
+			volumeName := "appframework-test-volume-" + testenv.RandomDNSName(3)
+			volumeSpec := []enterprisev1.VolumeSpec{testenv.GenerateIndexVolumeSpec(volumeName, testenv.GetS3Endpoint(), testenvInstance.GetIndexSecretName(), "aws", "s3")}
+
+			appSourceDefaultSpec := enterprisev1.AppSourceDefaultSpec{
+				VolName: volumeName,
+				Scope:   "cluster",
+			}
+			appSourceName := "appframework-" + testenv.RandomDNSName(3)
+			appSourceSpec := []enterprisev1.AppSourceSpec{testenv.GenerateAppSourceSpec(appSourceName, s3TestDir, appSourceDefaultSpec)}
+			appFrameworkSpec := enterprisev1.AppFrameworkSpec{
+				Defaults:             appSourceDefaultSpec,
+				AppsRepoPollInterval: 60,
+				VolList:              volumeSpec,
+				AppSources:           appSourceSpec,
+			}
+
+			// Create Single site Cluster and SHC, with App Framework enabled on SHC Deployer
+			// Deploy the CM
+			deployment.DeployClusterMaster(deployment.GetName(), "", "")
+
+			// Deploy the indexer cluster
+			indexerReplicas := 3
+			deployment.DeployIndexerCluster(deployment.GetName()+"-idxc", deployment.GetName(), indexerReplicas, deployment.GetName(), "")
+
+			// Deploy the SHC
+			shSpec := enterprisev1.SearchHeadClusterSpec{
```

Comment:
There is `CMREF` in the SH Spec. This SHC would not attach to Cluster Master.

Reply:
Are you referring to line 271-272? What to put instead?

Reply:
Never mind I missed it. Ignore this comment.